src_dualdesc/POLY_RecursiveDualDesc_MPI.h

```diff
@@ -350,10 +424,59 @@ vectface MPI_Kernel_DUALDESC_AdjacencyDecomposition(
     os << "nb_finished_oth=" << nb_finished_oth << "\n";
     return nb_finished_oth;
   };
+  auto short_wait = [&]() -> void {
+    int n_milliseconds = 10;
+    std::this_thread::sleep_for(std::chrono::milliseconds(n_milliseconds));
+  };
+
   auto wait = [&]() -> void {
     int n_milliseconds = 1000;
     std::this_thread::sleep_for(std::chrono::milliseconds(n_milliseconds));
   };
+  auto f_comm = [&](std::atomic_bool &done) -> void {
+    // start with no work
+    size_t cnt = 0;
+    while (!done && cnt < 10000) {
+      cnt++;
+      short_wait();
+    }
+    if(!done) 
+      os << "Start Probing" << std::endl;
+    // Start probing
+    while (!done) {
+        boost::optional<boost::mpi::status> prob = comm.iprobe();
+        if (prob) {
+          os << "process_mpi_status" << std::endl;
+          process_mpi_status(*prob);
```

Comment:
The `process_mpi_status` does process the database with RPL calls.
But the database is also accessed by the adjacency processing. So, we have potential
collisions here.
That is typically done by a lock. However, even that could not be sufficient.

Reply:
The thread is joined before the adjacency processing starts (the loop over theoutput). So this should not happen at the same time.

Reply:
Ok, I will examine it in detail tomorrow to convince myself that it is correct.

Reply:
Ok, now I am convinced that it is correct as indeed the operations are done separately.
That being said, the code has a weakness: The thread is constantly created and then destroyed.
I think performance-wise this is not great, because there is a cost to creating a thread and
destroying it. But maybe this interacts with the MPI stuff.

Anyway not blocking for merging but your comments are welcomed on that point.